File: mobile_deckz/lib/http_dao/game_history_dao.dart
import 'dart:convert';

import 'package:http/http.dart' as http;
import 'package:mobile_deckz/http_dao/http_dao.dart';
import 'package:mobile_deckz/http_dao/server_path.dart';

import '../model/game_history/game_history.dart';

class GameHistoryDao {
  static Future<GameHistory> getGameHistory() async {

    http.Response response = await HttpDao.get(ServerPath.gameHistoryPath);
    if (response.statusCode == 200) {
      Map<String, dynamic> json = jsonDecode(response.body);
      return GameHistory.fromJson(json);
    }
    return GameHistory(games: []);
  }
}

File: src/ulb/infof307/g01/gui/http/dao/GameHistoryDAO.java
package ulb.infof307.g01.gui.http.dao;

import com.google.gson.Gson;
import ulb.infof307.g01.model.gamehistory.GameHistory;
import ulb.infof307.g01.shared.constants.ServerPaths;

import java.io.IOException;
import java.net.http.HttpResponse;
import java.util.UUID;

public class GameHistoryDAO extends HttpDAO {
    /* ====================================================================== */
    /*                               DAO methods                              */
    /* ====================================================================== */

    public GameHistory getGameHistory() throws IOException, InterruptedException {
        HttpResponse<String> response = get(ServerPaths.GAME_HISTORY_PATH);
        checkResponseCode(response.statusCode());

        return new Gson().fromJson(response.body(), GameHistory.class);
    }

    public GameHistory getGameHistory(UUID deckId) throws IOException, InterruptedException {
        HttpResponse<String> response
                = get(ServerPaths.SPECIFIC_GAME_HISTORY_PATH + "?deck_id=" + deckId);
        checkResponseCode(response.statusCode());

        return new Gson().fromJson(response.body(), GameHistory.class);
    }
}
